src/Silverback.Integration/Messaging/Inbound/ErrorHandling/ErrorPoliciesHelper.cs

```diff
@@ -10,13 +10,32 @@ namespace Silverback.Messaging.Inbound.ErrorHandling
 {
     internal static class ErrorPoliciesHelper
     {
-        public static async Task<bool> ApplyErrorPoliciesAsync(ConsumerPipelineContext context, Exception exception)
+        public static async Task<bool> ApplyErrorPoliciesAsync(
+            ConsumerPipelineContext context,
+            Exception exception)
         {
-            var failedAttempts = context.Consumer.IncrementFailedAttempts(context.Envelope);
+            int failedAttempts = context.Consumer.IncrementFailedAttempts(context.Envelope);
 
             context.Envelope.Headers.AddOrReplace(DefaultMessageHeaders.FailedAttempts, failedAttempts);
+            context.Envelope.Headers.AddOrReplace(DefaultMessageHeaders.InvalidReason, exception.Message);
 
-            var errorPolicyImplementation = context.Envelope.Endpoint.ErrorPolicy.Build(context.ServiceProvider);
+            if (context.Envelope.Headers.Contains(DefaultMessageHeaders.TimestampKey))
+            {
+                context.Envelope.Headers.AddOrReplace(
```

Comment:
All these `x-source-*` headers are only relevant for the `MoveMessageErrorPolicy`, there's no need to set them every time and for all kind of policies.

Reply:
The headers have been moved to the `KafkaMessageHeaders` class.